packages/augur-core/source/contracts/reporting/RepPriceOracle.sol

```diff
@@ -0,0 +1,121 @@
+pragma solidity 0.5.10;
+
+import 'ROOT/IAugur.sol';
+import 'ROOT/libraries/Initializable.sol';
+import 'ROOT/uniswap/interfaces/IUniswapV2.sol';
+import 'ROOT/uniswap/interfaces/IUniswapV2Factory.sol';
+import 'ROOT/reporting/IRepPriceOracle.sol';
+
+
+contract RepPriceOracle is IRepPriceOracle, Initializable {
+
+    IAugur public augur;
+    address public cash;
+    uint256 public tau = 1 days / 15; // The time period (in blocks) of the twa, assuming a block is mined every 15 seconds. TODO: Review this value
+    IUniswapV2Factory public uniswapFactory;
+
+    struct ExchangeData {
+        IUniswapV2 exchange;
+        uint256 cashAmount;
+        uint256 repAmount;
+        uint256 blockNumber;
+        uint256 price;
+    }
+
+    mapping(address => ExchangeData) public exchangeData;
+
+    uint256 public gensisInitialRepPriceinAttoCash = 9 * 10**18;
+
+    function initialize(IAugur _augur) public beforeInitialized {
+        augur = _augur;
+        cash = _augur.lookup("Cash");
+        uniswapFactory = IUniswapV2Factory(_augur.lookup("UniswapV2Factory"));
+    }
+
+    // TODO: Consider when this should be called other than when the price is requested as part of new fee setting
+    function pokeRepPriceInAttoCash(IV2ReputationToken _reputationToken) external returns (uint256) {
+        if (exchangeData[address(_reputationToken)].exchange == IUniswapV2(0)) {
+            initializeUniverse(_reputationToken);
+        }
+        ExchangeData memory _newExchangeData = calculateNewExchangeData(_reputationToken);
+        exchangeData[address(_reputationToken)] = _newExchangeData;
+        return _newExchangeData.price;
+    }
+
+    function getRepPriceInAttoCash(IV2ReputationToken _reputationToken) external view returns (uint256) {
+        return calculateNewExchangeData(_reputationToken).price;
+    }
+
+    function calculateNewExchangeData(IV2ReputationToken _reputationToken) private view returns (ExchangeData memory) {
+        ExchangeData memory _exchangeData = exchangeData[address(_reputationToken)];
+        uint256 _blockNumber = block.number;
+        if (_blockNumber == _exchangeData.blockNumber) {
+            return _exchangeData;
+        }
```

Comment:
This can only happen if someone runs this query twice in one block, right? Is this going to happen often enough to justify the gas cost of a conditional vs occasionally running through the whole method?

Reply:
That's a good point. I feel like it is worth having in to make the conditions more readable and since this won't be called very often the slight additional cost savings isn't worth removing it.